pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -1617,6 +1650,54 @@ class StateVectorCudaMPI final
         mpi_manager_.Barrier();
     }
 
+    /**
+     * @brief collapse the state vector to a given basis state.
+     *
+     * @param wire_local Local wire index.
+     * @param branch Branch index.
+     */
+    void collapse_local_(const int wire_local, const bool branch) {
+        cudaDataType_t data_type;
+
+        if constexpr (std::is_same_v<CFP_t, cuDoubleComplex> ||
+                      std::is_same_v<CFP_t, double2>) {
+            data_type = CUDA_C_64F;
+        } else {
+            data_type = CUDA_C_32F;
+        }
+
+        std::vector<int> basisBits(1, wire_local);
+
+        double abs2sum0_local, abs2sum1_local;
+
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecAbs2SumOnZBasis(
+            /* custatevecHandle_t */ handle_.get(),
+            /* void *sv */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t nIndexBits */ BaseType::getNumQubits(),
+            /* double * */ &abs2sum0_local,
+            /* double * */ &abs2sum1_local,
+            /* const int32_t * */ basisBits.data(),
+            /* const uint32_t nBasisBits */ basisBits.size()));
+
+        auto abs2sum0 = mpi_manager_.allreduce<double>(abs2sum0_local, "sum");
+        auto abs2sum1 = mpi_manager_.allreduce<double>(abs2sum1_local, "sum");
+
+        double norm = (branch == 0) ? abs2sum0 : abs2sum1;
```

Comment:
```suggestion
        double norm = branch ? abs2sum0 : abs2sum1;
```
`branch` is already bool type